llama_index/indices/postprocessor/relevancy/base.py

```diff
@@ -0,0 +1,214 @@
+from llama_index.indices.postprocessor.types import BaseNodePostprocessor
+from enum import Enum
+
+"""
+post_processor_config: Dict = {
+    "node_splitter": { "splitter_type": "sentence" },
+    "relevancy_scorer": { 
+        "scorer_type": "sentence_transformer", 
+        "model_name": "paraphrase-distilroberta-base-v1" 
+        "agg_mode": "intra_node",
+    },
+    "filters": [
+        { 
+            "filter_type": "top_k", 
+            "top_k": 5, 
+            "agg_mode": "intra_node", 
+        }, 
+        { 
+            "filter_type": "marginalization", 
+            "agg_mode": "intra_node",
+            "embedding_function": "default_embedding",
+            "similarity_function": "cosine",
+        }
+    ],
+    # node_synthesizers have access to the old nodes and the new nodes
+    "synthesizer": {
+        "score_mode": "new_avg",
+        "node_synthesizers": [
+            { 
+                "synthesizer_type": "summarizer", 
+                "op_mode": "sub_node",
+                "summarizer_type": "hugging-face",
+                "model_name": "t5-base", 
+            },
+            # Highlight the top 1 sentence, used in 
+            # relevancy highlighting formatter
+            { 
+                "synthesizer_type": "relevancy_highlights",
+                "op_mode": "intra_node", 
+                "relevancy_mode": "top_k", 
+                "top_k": 1
+            },
+            { 
+                "synthesizer_type": "formatter", 
+                "op_mode": "intra_node", 
+                "format_mode": "join",
+                "join_on": "...",
+            },
+        ]
+    },
+}
+"""
+
+
+class BaseRelevancyPostprocessor(BaseNodePostprocessor):
+    """
+    Base class for relevancy post-processors.
+
+    It is meant to encompass the stages of:
+    - Splitting / Chunking
+    - Ranking Function / Relevancy Scoring
+    - Filtering / Marginalization
+    - Node Synthesis: Aggregation, Mapping
```

Comment:
if this is supposed to be a base class of a "relevancy postprocessor", would be good to have one concrete example implemented so we know the use case

Reply:
also why are all these steps within a node postprocessor? e.g. chunking and node synthesis?

Reply:
For pruning nodes of irrelevant sentences chunks.

Node synthesis doesn't necessarily mean LLM synthesis, it includes things like aggregating the score, generate metadata etc. along with LLM synthesis such as node-level summarization. Maybe a less confusing name would be better.

> if this is supposed to be a base class of a "relevancy postprocessor", would be good to have one concrete example implemented so we know the use case

Yes, I intended to add a flexible cross encoder reranking / pruning example based on sentence-transformers.